src/flashbar/__tests__/collapsible.test.tsx

```diff
@@ -45,297 +46,317 @@ const defaultProps = {
-  });
+        it('applies ARIA labels and title attributes to the item counter', () => {
+          const customLabels = {
+            errorIconAriaLabel: 'Custom error ARIA label',
+            successIconAriaLabel: 'Custom success ARIA label',
+            infoIconAriaLabel: 'Custom info ARIA label',
+            inProgressIconAriaLabel: 'Custom progress ARIA label',
+            warningIconAriaLabel: 'Custom warning ARIA label',
+          };
+          const flashbar = renderFlashbar({ i18nStrings: { ...customLabels } });
+          const innerCounter = findInnerCounterElement(flashbar);
+          for (const ariaLabel of Object.values(customLabels)) {
+            expect(innerCounter!.querySelector(`[aria-label="${ariaLabel}"]`)).toBeTruthy();
+            expect(innerCounter!.querySelector(`[title="${ariaLabel}"]`)).toBeTruthy();
+          }
+        });
 
-  describe('Sticky', () => {
-    it('scrolls the button into view when collapsing', () => {
-      scrollElementIntoViewMock.mockClear();
-      const flashbar = renderFlashbar();
-      findNotificationBar(flashbar)!.click();
-      findNotificationBar(flashbar)!.click();
-      expect(scrollElementIntoViewMock).toHaveBeenCalledTimes(1);
+        test.each([['success'], ['error'], ['info'], ['warning'], ['in-progress']] as FlashbarProps.Type[][])(
+          'item icon has aria-label from i18nStrings when no statusIconAriaLabel provided: type %s',
+          type => {
+            const wrapper = renderFlashbar({
+              i18nStrings: {
+                successIconAriaLabel: 'success',
+                errorIconAriaLabel: 'error',
+                infoIconAriaLabel: 'info',
+                warningIconAriaLabel: 'warning',
+                inProgressIconAriaLabel: 'in-progress',
+              },
+              items: [
+                {
+                  header: 'The header',
+                  content: 'The content',
+                  type: type === 'in-progress' ? 'info' : type,
+                  loading: type === 'in-progress',
+                },
+              ],
+            });
+
+            expect(wrapper.findItems()[0].find('[role="img"]')?.getElement()).toHaveAccessibleName(type);
+          }
+        );
+      });
+
+      describe('Sticky', () => {
+        it('scrolls the button into view when collapsing', () => {
+          scrollElementIntoViewMock.mockClear();
+          const flashbar = renderFlashbar();
+          findNotificationBar(flashbar)!.click(); // Expand
+          findNotificationBar(flashbar)!.click(); // Collapse
+          expect(scrollElementIntoViewMock).toHaveBeenCalledTimes(1);
+        });
+      });
     });
+  }
+
+  test('dismisses items', () => {
+    // Test this feature only without animations because TransitionGroup delays item removals by one frame.
+    // Customers should disable animations in their tests too:
+    // https://cloudscape.design/foundation/visual-foundation/motion/#implementation
+    disableMotion(true);
```

Comment:
If you manipulate with motion inside a single test, should the `beforeAll` hook convert to `beforeEach` to ensure the clean state between tests?

Reply:
Sorry I missed this comment. This test is outside the `describe` block that contains the `beforeAll` hook, so this call should not affect the tests in that block.

Reply:
This code contains a hidden danger, why not to resolve it by converting beforeAll to beforeEach

Reply:
Changed